js_modules/dagit/packages/core/src/resources/ResourceRoot.tsx

```diff
@@ -102,6 +140,12 @@ export const ResourceRoot: React.FC<Props> = (props) => {
     },
   });
 
+  React.useEffect(() => {
+    if (queryResult.data?.topLevelResourceDetailsOrError.__typename === 'ResourceDetails') {
+      setVerificationResult(queryResult.data.topLevelResourceDetailsOrError.verificationResult);
+    }
+  }, [queryResult]);
```

Comment:
Instead of having both the mutation and the query write to the same state, could the mutation just write to the same place in the Apollo cache that the query reads from? Then, if the mutation response successfully updates that object in cache, the query will pick it up right away and you don't need any `useState` management for it at all.

You might need to add an `id` to the `ResourceVerificationResult` type in order to do this cleanly, but that doesn't seem like a bad thing anyway.